src/Microsoft.Extensions.ProjectModel.Sources/MsBuild/MsBuildProjectContextBuilder.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.ProjectModel.Internal;
+using Microsoft.Extensions.FileProviders.Physical;
+
+namespace Microsoft.Extensions.ProjectModel
+{
+    internal class MsBuildProjectContextBuilder
+    {
+        private string _configuration;
+        private IFileInfo _fileInfo;
+        private string[] _buildTargets;
+        private Dictionary<string, string> _globalProperties = new Dictionary<string, string>();
+
+        public MsBuildProjectContextBuilder()
+        {
+            Initialize();
```

Comment:
The intent is to make it possible to override builder defaults. Overriding methods is easier than unwinding work done in the base constructor.